File: src/main/java/io/github/pixee/maven/operator/Command.java
package io.github.pixee.maven.operator;


import javax.xml.stream.XMLStreamException;
import java.io.IOException;
import java.net.URISyntaxException;

/**
 * Interface representing a command.
 */
public interface Command {
    /**
     * Given a context, performs an operation.
     *
     * @param pm Context (Project Model) to use.
     * @return true if the execution was successful AND the chain must end.
     */
    boolean execute(ProjectModel pm) throws URISyntaxException, IOException, XMLStreamException;

    /**
     * Post Processing, implementing a Filter Pattern.
     *
     * @param c ProjectModel for post-processing.
     * @return true if post-processing was successful.
     */
    boolean postProcess(ProjectModel c) throws XMLStreamException;
}

File: src/main/java/io/github/pixee/maven/operator/Command.kt
package io.github.pixee.maven.operator

import io.github.pixee.maven.operator.java.ProjectModelJ

/**
 * Represents a Command in a Chain of Responsibility Pattern
 */
interface Command {
    /**
     * Given a context, performs an operation
     *
     * @param pm Context (Project Model) to use
     * @return true if the execution was successful *AND* the chain must end
     */
    fun execute(pm: ProjectModelJ): Boolean

    /**
     * Post Processing, implementing a Filter Pattern
     */
    fun postProcess(c: ProjectModelJ): Boolean
}